### ISSUE TITLE
Components losts function names and crash when trying to enable /disable components

### ISSUE THREAD

**Issue by op2786** [Vector35/binaryninja-api#4177]
**Version and Platform (required):**
- Binary Ninja Version: 3.4.4189-dev
- OS: Windows 11, x64

**Bug Description:**
I've a BNDB from 3.4.4170 and when I open it in 3.4.4189 I saw that it lost function names in components. It crashes if I disable folders via Folders -> Show folder structure. 

When I reopen that binary in BN (with folders disabled), it still losees function names and when I try to enable folders it again crashes.

**Steps To Reproduce:**
* Find a BNDB that uses components and created with <= 3.4.4170
* Open it in 3.4.4189
* See function names lost
* Try disabling Folders -> Show folder structure in symbols menu
* See the crash

**Additional Information:**

This is the call stack for crash:

```
binaryninjaui!ComponentTreeSidebarWidgetType::createWidget+0x9e24
Qt6Core!QSortFilterProxyModel::columnCount+0x2a9
Qt6Core!QSortFilterProxyModel::columnCount+0x70
Qt6Widgets!QHeaderView::minimumSectionSize+0x111
Qt6Widgets!QHeaderView::initializeSections+0x38
Qt6Widgets!QHeaderView::reset+0x1e
Qt6Core!QObject::qt_static_metacall+0x195b
Qt6Core!QMetaObject::activate+0x84
Qt6Core!QAbstractItemModel::endResetModel+0x5b
Qt6Core!QSortFilterProxyModel::qt_static_metacall+0x17e
Qt6Core!QObject::qt_static_metacall+0x195b
Qt6Core!QMetaObject::activate+0x84
Qt6Core!QAbstractItemModel::endResetModel+0x5b
binaryninjaui!QFileAccessor::operator=+0x59cb
binaryninjaui!QFileAccessor::operator=+0x102f3
binaryninjaui!TagListModel::SetSectionSizeHints+0x23b2
binaryninjaui!CrossReferenceWidget::wheelEvent+0x1af8
binaryninjacore!BNGetLowLevelSSARegistersWithoutVersions+0x21804
binaryninja+0x447f4
```

I can share the BNDB and crash dump in slack.

**Comment by ccarpenter04** [comment]
I also have been experiencing this issue on 3.4.4189-dev, seems to be very recent.

**Comment by bambu** [comment]
Not sure if related to this, but also appears to crash when opening a binary for analysis in a new tab after a bndb is open in another tab. Tried on both macos and ubuntu on 3.4.4189-dev. Disabling components view fixes the issue.

**Comment by cxnder** [comment]
Resolved in **builds >= 3.4.4199**
